account-kit/rn-signer/example/src/screens/otp-auth.tsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable import/extensions */
+import type { User } from "@account-kit/signer";
+import { useEffect, useState } from "react";
+import {
+  View,
+  Text,
+  TextInput,
+  StyleSheet,
+  TouchableOpacity,
+} from "react-native";
+
+import Config from "react-native-config";
+import { RNAlchemySigner } from "@account-kit/react-native-signer";
+export default function MagicLinkAuthScreen() {
+  const [email, setEmail] = useState<string>("");
+  const [user, setUser] = useState<User | null>(null);
+
+  const signer = RNAlchemySigner({
+    client: { connection: { apiKey: Config.API_KEY! } },
+  });
+
+  const [awaitingOtp, setAwaitingOtp] = useState<boolean>(false);
+
+  const [otpCode, setOtpCode] = useState<string>("");
+
+  const handleUserAuth = ({ otpCode }: { otpCode: string }) => {
+    setAwaitingOtp(false);
+    signer
+      .authenticate({
+        otpCode: otpCode,
+        type: "otp",
+      })
+      .then((res) => {
+        console.log("res", res);
+        setUser(res);
+      })
+      .catch(console.error);
+  };
+
+  useEffect(() => {
+    // get the user if already logged in
+    signer.getAuthDetails().then(setUser);
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'signer'. Either include it or remove the dependency array.
<!-- __reviewdog__:ChA5NzQ0YzU3MjY0NmQ4M2NiEgZyZGpzb24= -->